### DISCUSSION TITLE
Which is the purpose of the _GAME macro?

### DISCUSSION THREAD

**Discussion by nicola907** [climateinteractive/SDEverywhere#332] [Q&A] [answered]
Hi everyone. 
I have generated the C code of my model using the following command:

    `sde build mymodel.mdl --spec mymodel.json`

Then I have noticed that in the generated "vensim.h" file there is a macro defined like this:

    `#define _GAME(x) (x)`

It simply returns the value that I pass to it. The problem is that in the generated "mymodel.c" file, more specifically the "evalAux0" function, this macro is used like this:

    `_myvariable = _GAME(0.0);`

It sets "myvariable" to the value 0.0. The problem is that I don't want it to be always 0.0 because it's one of my inputs and the user must be able to give other values.

**Comment by chrispcampbell** [comment, marked as answer]
Hi @nicola907, apologies for the long overdue reply.  I wanted to let you know that just this month we finished implementing full support for the Vensim `GAME` function and gaming mode in general in SDEverywhere.

Back when you wrote this message last year, SDEverywhere did not have full support for the `GAME` function, and the documentation noted that it was implemented as a no-op, meaning that it would only return the input (`x`) value.  But now that we've implemented full support for the `GAME` function, it is now possible to supply user-provided gaming inputs to the model at runtime.

More details are available in the [August 2024 Updates](https://github.com/climateinteractive/SDEverywhere/discussions/513) announcement.  Hope this is useful to you and others who have been requesting gaming support in SDEverywhere.
